api-tests/src/utils/jar-generator.ts

```diff
@@ -37,27 +37,36 @@ export const generateJarPayload = async (
       sub: session.subject,
       govuk_signin_journey_id: session.journeyId,
       state: getRandomString(16),
-      redirect_uri: config.orch.redirectUrl,
+      redirect_uri: session.redirectUrl || config.orch.redirectUrl,
```

Comment:
I wonder if we should make the overrides here cleverer - e.g. if you had:
```
redirect_uri: payloadData.redirect_uri || config.orch.redirectUrl
```
So if you have a payload JSON with `"redirect_uri": ""` then it'll get populated automatically, but if you have `"redirect_uri": "invalid"` then it won't.

(That would save you passing the url around everywhere, and it'd work more like the invalid audience)

Reply:
good idea!